File: scripts/lib/slug.rb
def slugify(value)
  value = value.downcase.strip
  value = value.gsub(/(?<!\s)(?!\w)'(?=\w)/, "-")
  value = value.gsub(/\s\&\s/, " ")
  value = value.gsub(/[\'\"\(\)]/, "")
  value = value.gsub(/\.\s/, "-")
  value = value.gsub(/[\s\,\.\&]{1}/, '-')
  value = value.gsub("ä", "a").gsub("ö", "o").gsub("ü", "u").gsub("á", "a")
  value = value.gsub("+", "-plus")
end

File: src/lib/slug.ts
/** Port of scripts/lib/slug.rb */
export function slugify(value: string): string {
  return value
    .toLowerCase()
    .trim()
    .replace(/(?<!\s)(?!\w)'(?=\w)/g, "-")
    .replace(/\s&\s/g, " ")
    .replace(/['\"()]/g, "")
    .replace(/\.\s/g, "-")
    .replace(/[\s,.&]/g, "-")
    .replace(/ä/g, "a")
    .replace(/ö/g, "o")
    .replace(/ü/g, "u")
    .replace(/á/g, "a")
    .replace(/\+/g, "-plus");
}
